tracer/resources/processes.py

```diff
@@ -194,8 +196,23 @@ def files(self):
 
 		# Process arguments
 		for arg in self.cmdline()[1:]:
-			if os.path.isfile(arg):
-				files.append(arg)
+			if not os.path.isabs(arg):
+				continue
+
+			"""
+			Process arguments could be referring to files on remote filesystems and
+			os.path.isfile will hang forever if the shared FS is offline.
+			Instead, use a subprocess that we can time out if we can't reach some file.
+			"""
+			process = Popen(['test', '-f', arg], stdout=PIPE, stderr=PIPE)
+			timer = Timer(0.5, process.kill)
+			try:
+				timer.start()
+				process.communicate()
+				if process.returncode == 0:
+					files.append(arg)
+			finally:
+				timer.cancel()
```

Comment:
Looks good to me @jturel, thank you!
In my opinion, it would be a good idea to wrap this into its own function. It is a non-trivial piece of code, you even added a helpful docstring ... I think it would deserve it to be separated.

What do you think?

Reply:
Good idea, will make the change!

Reply:
@FrostyX updated. let me know if you think the `safe_isfile` is in the correct place, or if it could be moved to a better location.

Reply:
Looks good to me, keep it there :-)